assets/src/edit-story/utils/nativeCopyPasteExpected.js

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Checks if the native copy paste should proceed,
+ * either if text is selected or if there's focus on a text/number input.
+ *
+ * @return {boolean} If native handling is expected.
+ */
+function nativeCopyPasteExpected() {
+  const { activeElement } = document;
+  const { tagName, type, isContentEditable } = activeElement;
+
+  // If it's an input in focus, do native handling.
+  if (
+    'textarea' === tagName.toLowerCase() ||
+    (('input' === tagName.toLowerCase() || 'true' === isContentEditable) &&
+      ('text' === type || 'number' === type))
+  ) {
+    return true;
+  }
+
+  const selection = window.getSelection();
```

Comment:
But what if the `window.getSelection` is not available? The code in the `getCaretCharacterOffsetWithin` says that it's a real possibility. I'm not sure if that's true or not.

Reply:
I'm quite certain that `window.getSelection` should always be available in this case, it has full cross-browser support. Perhaps `getCaretCharacterOffsetWithin` is considering the option of the element having been unmounted meanwhile since it's getting the selection based on a specific node.

Additionally, `getCaretCharacterOffsetWithin` is supporting browsers that we don't actually support -- e.g. `element.document` and `doc.parentWindow` both are for IE which we don't really support generally.